api.d.ts

```diff
@@ -1,80 +1,115 @@
 /********************************************************
- JobRouter JavaScript-API Stubs v5.1
+ JobRouter JavaScript-API Stubs v2022.1
  ********************************************************/
 type CustomErrorHandler = (errorMessage: string) => void;
-type ElementId = string;
+type DecimalSeparator = "" | "," | ".";
+type DialogFunctionUserParameters = {
+    [key: string]: unknown;
+};
+type DialogSuccessReturnObject = {
+    status: string;
+    result?: {
+        [key: string]: unknown;
+    };
+};
+type DialogErrorReturnObject = {
+    status: string;
+    message?: string;
+    result?: {
+        [key: string]: unknown;
+    };
+};
+type ElementValue = string | number | Date | boolean;
+type MessageData = {
+    [key: string]: ElementValue;
```

Comment:
I had used `ElementValue` since it is the return type of `jr_get_value` which is used in the example in the documentation, but after reading more into it I think this is wrong.

The contents of jobrouter.js for this function are as follows:

```js
// Message - Functions
// ====================

var jr_message = function(message, data, customHandler) {
    var retValue = "";
    if (window.$JRMESSAGE[message]) {
        if (data) {
            retValue = window.$JRMESSAGE[message];
            $H(data).each(function(a) {
                $H(data).each(function(a) {
                    // Eckige Klammer notation ist nicht mehr offiziell dokumentiert,
                    // wird aber weiter unterstÃ¼tzt um alte Prozesse lauffÃ¤hig zu halten.
                    retValue = retValue.replace("[" + a.key + "]", a.value);
                    retValue = retValue.replace("#{" + a.key + "}", a.value);
                });
            });
        } else {
            retValue = window.$JRMESSAGE[message];
        }

        alert(jQuery("<div />").html(retValue).text());

    } else {
        throw new window.JobRouterException("No message '" + message + "' found!", customHandler);
    }
};
```

This implies to me that maybe only `string` types should be accepted, since it's concatenating to other strings, and the documentation for `.replace` from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace indicates it should be a string

The JobRouter documentation reads:

> Optional: Object containing parameters to be substituted in the text.
> Example for filling place-keepers [var1] and [var2] in a process message:
> `{var1:"Wert1", var2:"Wert2"} or {var1:"Wert1", var2:jr_get_value("limit")}`

This shows using `jr_get_value("limit")` which can return a string, but can also return other types, which would need to be handled (converted to a string) in order to work as expected.

I've used `string` in https://github.com/jobrouter/javascript-api/pull/4/commits/253d2ce7fe9045f52d93ef3abead156924668c94